File: 1750-minimum-length-of-string-after-deleting-similar-ends/1750-minimum-length-of-string-after-deleting-similar-ends.java
class Solution {
    public int minimumLength(String s) {
        int i=0;
        int j=s.length()-1;
        while(i<j){
            if (s.charAt(i)==s.charAt(j)){
                while(i+1<j && s.charAt(i)==s.charAt(i+1)) i++;
                while(i<j-1 && s.charAt(j)==s.charAt(j-1)) j--;
            }
            else return j-i+1;
            i+=1;
            j-=1;
        }
        return j-i+1;
    }
}

File: 1750-minimum-length-of-string-after-deleting-similar-ends/1750-minimum-length-of-string-after-deleting-similar-ends.py
class Solution:
    def minimumLength(self, s: str) -> int:
        # k=len(s)
        i=0
        j=len(s)-1
        # mt=''
        while(i<j):
            if s[i]==s[j]:
                while(i+1<j and s[i]==s[i+1]):
                    i+=1
                while(i<j-1 and s[j]==s[j-1]):
                    j-=1
                i+=1
                j-=1
            else:
                return j-i+1
        return j-i+1
                    
        